phoenix-core/src/main/java/org/apache/phoenix/expression/visitor/CloneExpressionVisitor.java

```diff
@@ -107,7 +109,9 @@ public Expression visitLeave(ComparisonExpression node, List<Expression> l) {
 
     @Override
     public Expression visitLeave(LikeExpression node, List<Expression> l) {
-        return Determinism.PER_INVOCATION.compareTo(node.getDeterminism()) > 0 ? node :  new LikeExpression(l);
```

Comment:
No instanceof checks. Add an abstract clone method to LikeExpression and call that here instead.

    abstract public LikeExpression clone(List<Expression> children);